### DISCUSSION TITLE
List of String type enforcement and default

### DISCUSSION THREAD

**Discussion by ekarlso** [cuelang/cue#400] [General] [answered]
Can I somehow specify foo: []string | *[commonName]

*Usecase*
We want to specify something like: 
```
certificate: [ID=_]: cm_alpha3.#Certificate & {
	kind:       cm_alpha3.#CertificateKind
	apiVersion: "cert-manager.io/v1alpha3"
	metadata: name: ID

	spec: issuerRef: {
		name: string | *"letsencrypt-prod"
		kind: string | *cm_alpha3.#ClusterIssuerKind
	}

	spec: {
		secretName: string | *ID
		commonName: string | <5
        renewBefore: string | *"360h0m0s"
        dnsNames: string | * [commonName]
	}
}
```

And have the dnsNames be default to [commonName] if dnsNames is not passed. Is this doable?

**Comment by gogommartin** [comment, marked as answer]
I'm new to Cue myself so this may not be correct, but I believe what you're looking for is:
```
dnsNames: [...string] | *[commonName]
```

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/400.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
